dask/array/core.py

```diff
@@ -2374,11 +2385,37 @@ def store(x, index, lock):
     slices = slices_from_chunks(arr.chunks)
 
     name = 'store-%s' % arr.name
-    dsk = dict(((name,) + t[1:], (store, t, slc, lock))
+    dsk = dict(((name,) + t[1:], (store, t, slc, lock, region))
                for t, slc in zip(core.flatten(arr._keys()), slices))
     return dsk
 
 
+def storeindask(out, index, x):
+    #Loop through output chunks and setitem:
+    out_sliced = out[index]
+    for slc in slices_from_chunks(out_sliced.chunks):
+        dsk = optimize(out_sliced[slc].dask, out_sliced[slc]._keys())
+        if len(dsk) == 1:
+            for key in dsk:
+                if base_condition(dsk, key):
+                    operator.setitem(dsk[key][1], dsk[key][2], x[slc])
+        elif len(dsk) == 2:
+            for key in dsk:
+                if base_condition(dsk, key):
+                    operator.setitem(dsk[dsk[key][1]], dsk[key][2], x[slc])
+        else:
+            raise NotImplementedError('Only concantenated dask arrays can be stored to')
+
+
+def base_condition(dsk, key):
+    #Simple condition that allows simple setitem:
+    return (isinstance(key, tuple) and
+            isinstance(dsk[key],tuple) and
+            len(dsk[key]) == 3 and
+            dsk[key][0] is getarray and
+            isinstance(dsk[key][2], tuple))
```

Comment:
I think that we don't.  I think that we should drop support for writing to dask.arrays.  I think that there will be many small problems like this in the future that will be difficult to support.